src/main.rs

```diff
@@ -1376,13 +1377,126 @@ fn parse_manpage_at_path(
+fn parse_and_output_man_pages<P: AsRef<Path>>(
+    paths: impl Iterator<Item = P>,
+    output_directory: P,
+    Progress(show_progress): Progress,
+    deroff_only: bool,
+    write_to_stdout: bool,
+) -> Result<(), String> {
+    let mut paths = paths
+        .map(|path| path.as_ref().to_owned())
+        .collect::<Vec<_>>();
+    paths.sort();
+    paths.dedup();
+
+    let total_count = paths.len();
+
+    let (mut successful_count, mut index) = (0, 0);
+
+    let mut cmd_name = "".to_owned();
+
+    // Get the number of digits in total_count
+    let padding_len = (total_count as f32).log(10.) as usize + 1;
+
+    let mut last_progress_string_length = 0;
+
+    if show_progress && !write_to_stdout {
+        println!(
+            "Parsing man pages and writing completions to {:?}",
+            output_directory.as_ref()
+        );
+    }
+
+    // As far as I can tell, we're fully supporting xz / lzma, so the checks here aren't needed
+
+    for manpage_path in paths {
+        index += 1;
+        // foo/bar/gcc.1.gz -> gcc.1.gz
+        // TODO: Expect
+        let man_file_name = manpage_path
+            .file_name()
+            .expect("Failed to get name of manpage path")
+            .to_string_lossy()
+            .to_mut()
+            .to_owned();
+
+        // gcc.1.gz -> gcc
+        cmd_name = match man_file_name.split('.').next() {
+            Some(cmd_name) => cmd_name.to_owned(),
+            None => return Err(format!("Failed to get cmd_name from {}", man_file_name)),
+        };
+
+        if show_progress {
+            // len(str(index))
+            let cur_len = (index as f32).log(10.) as usize + 1;
+            let progress_str = format!(
+                "  {} / {} : {}",
+                // str(index).rjust(padding_len)
+                format!(
+                    "{}{}",
+                    (0..(padding_len - cur_len)).fold("".to_owned(), |mut acc, _| {
+                        acc.push_str(" ");
+                        acc
+                    }),
+                    index
+                ),
+                total_count,
+                man_file_name,
+            );
+
+            // padded_progress_str = progress_str.ljust(last_progress_string_length)
+            let padded_progress_str = format!(
+                "{}{}",
+                progress_str,
+                (0..(last_progress_string_length - progress_str.len())).fold(
+                    "".to_owned(),
+                    |mut acc, _| {
+                        acc.push_str(" ");
+                        acc
+                    }
+                )
+            );
+
+            last_progress_string_length = progress_str.len();
+            // TODO: Expects
+            let stdout = std::io::stdout();
+            let mut lock = stdout.lock();
+            lock.write(format!("\r{}\r", padded_progress_str).as_bytes())
+                .expect("Failed to write to stdout");
+
+            lock.flush().expect("Failed to flush stdout");
```

Comment:
Do we need flush? I thought it automatically flush when it goes out of scope? Is this good (not formatted)?
```rust
            io::stdout().write(format!("\r{}\r", padded_progress_str).as_bytes())
                .expect("Failed to write to stdout");
```

Reply:
I'm not sure. @scooter-dangle do you know?

Reply:
Let me ping @lonami, I wonder if he knows.

Reply:
Not sure, the examples in the documentation for [`stdout`](https://doc.rust-lang.org/stable/std/io/fn.stdout.html) don't mention anything. Unless I'm looking in the wrong place perhaps the docs could be updated to note the behaviour. https://github.com/rust-lang/rust/issues/23205 makes me thing it'll be flushed.

Reply:
I think the `flush` question is interesting in general, but in this case, is there any reason for us to not just `print!` or `println!` instead of explicitly manipulating `stdout`?

Reply:
Seems like we do need to flush. Marking as resolved.